cms/djangoapps/modulestore_migrator/tasks.py

```diff
@@ -89,7 +89,7 @@ class _MigrationContext:
     Context for the migration process.
     """
     existing_source_to_target_keys: dict[  # Note: It's intended to be mutable to reflect changes during migration.
-        UsageKey, PublishableEntity
```

Comment:
The underlying bug was here. It was only considering the last migration, so after migrating the second time, the slug used for the first migration would appear available and reused (causing an update).